### PR TITLE
Migrate TAPI performance tests to use the Gradle profiler infrastructure

### PR DESCRIPTION
**Pull request by wolfs**
Fixes gradle/gradle-private#3189
Fixes https://github.com/gradle/gradle-private/issues/3202

### CODE REVIEW

**Comment by wolfs** [comment]
I was able to migrate the TAPI scenarios to Gradle profiler, though it seems like when using Gradle profiler the baseline version is faster. For `get IDE model on largeMonolithicJavaProject for Eclipse` the difference seems to be 50% (30ms).

I don't really understand why this is the case. I tried reversing the order of runs, though that didn't make a difference.

I wonder if the problem is similar to what we see in `create many deprecation warnings`, which is currently disabled since it shows a regression of about 20ms right after rebaselining, aka with the same code.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleProfilerBuildExperimentRunner.java [hunk 1]
```diff
@@ -53,12 +54,22 @@ public AbstractGradleProfilerBuildExperimentRunner(BenchmarkResultCollector resu
...
             ? ProfilerFlameGraphGenerator.NOOP
             : new JfrFlameGraphGenerator(new File(jfrProfileTargetDir));
-        this.profiler = jfrProfileTargetDir == null
-            ? org.gradle.profiler.Profiler.NONE
-            : ProfilerFactory.of(Arrays.asList("jfr")).createFromOptions(new OptionParser().parse());
+        this.profiler = createProfiler(jfrProfileTargetDir);
         this.resultCollector = resultCollector;
     }
 
+    private org.gradle.profiler.Profiler createProfiler(String jfrProfileTargetDir) {
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleProfilerBuildExperimentRunner.java:61]: I already merged this change to `master`, you can rebase it.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossVersionPerformanceTestRunner.groovy [hunk 2]
```diff
@@ -54,6 +54,7 @@ class AbstractCrossVersionPerformanceTestRunner extends PerformanceTestSpec {
     String testProject
     File workingDir
     boolean useDaemon = true
+    boolean useToolingApi = false
```
Comment (blindpirate) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossVersionPerformanceTestRunner.groovy:57]: I love the neat solution of reusing it

**Review by blindpirate — COMMENTED** [review]
This LGTM now, if everything is green then all good.

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleProfilerBuildExperimentRunner.java (hunk 1 above)
Comment (wolfs) [reply subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractGradleProfilerBuildExperimentRunner.java]: Rebased.

**Comment by wolfs** [comment]
Looks like the performance tests only start to fail after 200 warmups. Then the baseline version starts to keep the lower threshold (around 60ms), while the current version alternates between the lower and upper measurements (100ms). Maybe that is related that the current distribution lives in a location with a longer path?

**Comment by wolfs** [comment]
The reason for the failing performance test was the length of the path to the Gradle installation under test after all: https://github.com/gradle/gradle/pull/14438/commits/f37a9a84d6e35a3ebe899b213b9268f9e4c1e211

Now all tests are passing. We should probably merge this PR after https://github.com/gradle/gradle/pull/14441.

Then we should be able to remove the whole `Profiler` infrastructure from the Gradle build.

**Comment by wolfs** [comment]
@blindpirate PTAL!

**Pull request merged**